numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2923,8 +2923,36 @@ static PyObject *
 
     /* TODO: include type name in error message, which is not @name@ */
     PyObject *obj = NULL;
+#if defined(_@TYPE@_IS_CDOUBLE) || defined(_@TYPE@_IS_CFLOAT)
+    static char *kwnames[] = {"", "", NULL};  /* positional-only */
+    static char *parse_error = "Could not convert arguments into a complex scalar. '%R' given.";
+    PyObject *real_obj = NULL;
+    PyObject *imag_obj = NULL;
+    if (PyArg_ParseTupleAndKeywords(args, kwds, "|OO", kwnames, &real_obj, &imag_obj)) {
+        if (imag_obj == NULL) {
+            obj = real_obj;
+            Py_XINCREF(obj);
+        }
+        else if (PyNumber_Check(real_obj) && PyNumber_Check(imag_obj)) {
+            if (PyComplex_Check(real_obj) || PyComplex_Check(imag_obj)) {
```

Comment:
OK, without this we would allow `np.complex128(numpy_complex, numpy_complex)`, although it would give a warning.  I don't have a strong opinion on the need, I agree, this is mainly about accepting the trivial case of `np.complex128(3., 4)` with python scalars anyway.